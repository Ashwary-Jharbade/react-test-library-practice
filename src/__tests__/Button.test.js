import { act, create } from "react-test-renderer";
import ButtonComp from "../components/Button";

test("Testing if ButtonComp component rendered", () => {
  const buttonComponent = create(<ButtonComp />);
  const tree = buttonComponent.toJSON();
  expect(tree).toBeTruthy();
});

test("Testing if ButtonComp component rendered with Click prop", () => {
  const buttonComponent = create(<ButtonComp label="Click" />);
  const tree = buttonComponent.toJSON();
  const { props } = buttonComponent.root;
  expect(tree).toBeTruthy();
  expect(props.label).toBe("Click");
});

describe("Button compoenent testing", () => {
  test("Testing data change on button click", () => {
    const buttonComponent = create(<ButtonComp label="Click" />);
    const { root } = buttonComponent;
    const buttonInstance = root.findByProps({ "data-testId": "btn1" });
    expect(root.props.label).toBe("Click");
    act(() => {
      buttonInstance.props.onClick();
    });
    expect(buttonInstance.props.children).toBe("Changed");
    expect(buttonComponent.toJSON()).toMatchSnapshot();
  });

  test("Testing for specific button", () => {
    const buttonComponent = create(<ButtonComp label="Click" />);
    const instance = buttonComponent.root;
    const buttonInstance = instance.findByProps({ "data-testId": "btn2" });
    expect(buttonInstance.props.children).toBe("Click");
    expect(buttonComponent.toJSON()).toMatchSnapshot();
  });
});
