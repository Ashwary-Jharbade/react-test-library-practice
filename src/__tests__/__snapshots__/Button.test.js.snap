// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button compoenent testing Testing data change on button click 1`] = `
Array [
  <button
    data-testId="btn1"
    onClick={[Function]}
    type="button"
  >
    Changed
  </button>,
  <button
    data-testId="btn2"
    type="button"
  >
    Click
  </button>,
]
`;

exports[`Button compoenent testing Testing for specific button 1`] = `
Array [
  <button
    data-testId="btn1"
    onClick={[Function]}
    type="button"
  >
    Click
  </button>,
  <button
    data-testId="btn2"
    type="button"
  >
    Click
  </button>,
]
`;
